bgpd/bgp_updgrp_packet.c

```diff
@@ -1142,7 +1142,12 @@ void subgroup_default_update_packet(struct update_subgroup *subgrp,
 
 	(void)bpacket_queue_add(SUBGRP_PKTQ(subgrp), s, &vecarr);
 	subgroup_trigger_write(subgrp);
-	subgrp->scount++;
+
+	if (!CHECK_FLAG(subgrp->sflags,
+			SUBGRP_STATUS_PEER_DEFAULT_ORIGINATED)) {
```

Comment:
Can't we rely on `SUBGRP_STATUS_DEFAULT_ORIGINATE`?

For instance existing code like:
```c
	if (withdraw) {
		/* Withdraw the default route advertised using default
		 * originate
		 */
		if (CHECK_FLAG(subgrp->sflags, SUBGRP_STATUS_DEFAULT_ORIGINATE))
			subgroup_default_withdraw_packet(subgrp);
		UNSET_FLAG(subgrp->sflags, SUBGRP_STATUS_DEFAULT_ORIGINATE);

```
If we move `UNSET_FLAG(subgrp->sflags, SUBGRP_STATUS_DEFAULT_ORIGINATE);` to `subgroup_default_withdraw_packet` and do the math on `scount` according to this flag?

The same for `subgroup_default_update_packet`.

Reply:
I think we should have a separate flag to take care of the SNT counters,
since in order to handle the UPDATE scenarios, we end up unsetting the flag `SUBGRP_STATUS_DEFAULT_ORIGINATE` at multiple places. So, in that case, if we still end up relying on this particular flag, there might be cases where we will end up with inconsistent SNT counters.

Reply:
For instance, suppose we already have default route originate through `neighbor x.x.x.x default-originate` cmd,
now we go ahead and associate a route-map with it, so in this case, we will already have the `SUBGRP_STATUS_DEFAULT_ORIGINATE` flag unset by the time we reach `subgroup_default_update_packet`, to trigger a new update, and that would again lead to incrementing the SNT counter.
Please correct me if i am wrong.